readthedocs/search/faceted_search.py

```diff
@@ -247,14 +260,92 @@ def query(self, search, query):
 
     def _get_script_score(self):
         """
-        Gets an ES script to map the page rank to a valid score weight.
+        Gets an ES script that combines the page rank and views into the final score.
+
+        **Page ranking weight calculation**
+
+        Each rank maps to a element in the ranking list.
+        -10 will map to the first element (-10 + 10 = 0) and so on.
+
+        **Page views weight calculation**
+
+        We calculate two values:
+
+        - absolute: this is equal to ``log10(views + 1)``
+          (we add one since logarithms start at 1).
+          A logarithmic function is a good fit due to its growth rate.
```

Comment:
Nope, here views is the number of views from that page in that version, this value isn't compared to anything (that's why it's absolute).

Reply:
So, if we have a page `/api/v2.html` that has a lot of views in version `1.0`, then a new version of the docs is published, `1.1`; the new page `/1.1/api/v2.html` won't be shown first in the results because it will have less views?

If that is correct, won't we be pointing the user always to old pages because they will have more views than the up-to-date version of the exact same page?

Reply:
Similar example searching for the term `API` in `1.1` version of the docs. It may refers to another page (i.e. `configuration.html`) instead of `api/v2.html` because the newer version of `configuration.html` has more views than the newer version of `api/v2.html` --but historically (old versions) `api/v2.hml` has a lot more views. Makes sense?